token/cli/src/main.rs

```diff
@@ -1891,6 +1911,164 @@ async fn command_sync_native(
     })
 }
 
+// Both enable-required_transfer-mesos and disable-required_transfer-mesos, switches with enable bool
+async fn command_required_transfer_memos(
+    token_account_address: Pubkey,
+    owner: Pubkey,
+    bulk_signers: BulkSigners,
+    enable: bool,
+    config: &Config<'_>,
+) -> CommandResult {
+    let program_id = if config.sign_only {
+        config.program_id
+    } else {
+        config
+            .rpc_client
+            .get_account(&token_account_address)
+            .await
+            .map_err(|err| {
+                format!(
+                    "Token account {} does not exist: {}",
+                    token_account_address, err
+                )
+            })?
+            .owner
+    };
+    let mut instructions: Vec<Instruction> = Vec::new();
+    let reallocate_instruction = reallocate_if_needed(
+        &program_id,
+        &token_account_address,
+        &owner,
+        vec![ExtensionType::MemoTransfer],
+        config,
+    )
+    .await?;
+    if let Some(instr) = reallocate_instruction {
+        instructions.push(instr);
+    }
+    if enable {
+        instructions.push(
+            spl_token_2022::extension::memo_transfer::instruction::enable_required_transfer_memos(
+                &program_id,
+                &token_account_address,
+                &owner,
+                &config.multisigner_pubkeys,
+            )?,
+        );
+    } else {
+        instructions.push(
+            spl_token_2022::extension::memo_transfer::instruction::disable_required_transfer_memos(
+                &program_id,
+                &token_account_address,
+                &owner,
+                &config.multisigner_pubkeys,
+            )?,
+        );
+    }
+    let tx_return = handle_tx(
+        &CliSignerInfo {
+            signers: bulk_signers,
+        },
+        config,
+        false,
+        0,
+        instructions,
+    )
+    .await?;
+    Ok(match tx_return {
+        TransactionReturnData::CliSignature(signature) => {
+            config.output_format.formatted_string(&signature)
+        }
+        TransactionReturnData::CliSignOnlyData(sign_only_data) => {
+            config.output_format.formatted_string(&sign_only_data)
+        }
+    })
+}
+
+// Checks if reallocation needed for new specified extensions. If needed returns reallocate instruction
+async fn reallocate_if_needed(
+    program_id: &Pubkey,
+    token_account_pubkey: &Pubkey,
+    owner_pubkey: &Pubkey,
+    new_extension_types: Vec<ExtensionType>,
+    config: &Config<'_>,
+) -> Result<Option<Instruction>, Error> {
+    let token_account = config
+        .rpc_client
+        .get_token_account(token_account_pubkey)
+        .await?
+        .expect("Invalid token account pubkey");
```

Comment:
as noted above, whether we keep the function or not, we shouldnt fetch

Reply:
resolved with d37c851